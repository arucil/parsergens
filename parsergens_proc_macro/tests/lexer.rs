
use insta::assert_debug_snapshot;

parsergens_proc_macro::parsergen!(mod parser: LALR("fixtures/expr_tokens.pg"));

#[test]
fn simple() {
  let tokens = parser::lex(r"(3.2 * 51 + Foo_1) / 20. -5  ,    ").collect::<Vec<_>>();
  assert_debug_snapshot!(tokens);
}