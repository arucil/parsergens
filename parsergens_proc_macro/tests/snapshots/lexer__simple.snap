---
source: parsergens_proc_macro/tests/lexer.rs
expression: tokens
---
[
    Ok(
        Token {
            kind: Lparen,
            text: "(",
            start: 0,
            end: 1,
        },
    ),
    Ok(
        Token {
            kind: Number,
            text: "3.2",
            start: 1,
            end: 4,
        },
    ),
    Ok(
        Token {
            kind: Mul,
            text: "*",
            start: 5,
            end: 6,
        },
    ),
    Ok(
        Token {
            kind: Number,
            text: "51",
            start: 7,
            end: 9,
        },
    ),
    Ok(
        Token {
            kind: Plus,
            text: "+",
            start: 10,
            end: 11,
        },
    ),
    Ok(
        Token {
            kind: Ident,
            text: "Foo_1",
            start: 12,
            end: 17,
        },
    ),
    Ok(
        Token {
            kind: Rparen,
            text: ")",
            start: 17,
            end: 18,
        },
    ),
    Ok(
        Token {
            kind: Div,
            text: "/",
            start: 19,
            end: 20,
        },
    ),
    Ok(
        Token {
            kind: Number,
            text: "20.",
            start: 21,
            end: 24,
        },
    ),
    Ok(
        Token {
            kind: Minus,
            text: "-",
            start: 25,
            end: 26,
        },
    ),
    Ok(
        Token {
            kind: Number,
            text: "5",
            start: 26,
            end: 27,
        },
    ),
    Ok(
        Token {
            kind: Comma,
            text: ",",
            start: 29,
            end: 30,
        },
    ),
]
